Tasks/AzureRmWebAppDeployment/kuduutility.ts

```diff
@@ -148,21 +148,26 @@ export async function ensurePhysicalPathExists(publishingProfile, physicalPath:
     return defer.promise;
 }
 
-export async function runPostDeploymentScript(publishingProfile, physicalPath, scriptType, inlineScript, scriptPath, appOfflineFlag) {
+export async function runPostDeploymentScript(publishingProfile, physicalPath, scriptType, inlineScript, scriptPath, appOfflineFlag, isLinux = false) {
     var scriptFile = getPostDeploymentScript(scriptType, inlineScript, scriptPath);
     var uniqueID = azureUtility.generateDeploymentId();
     tl.debug('Deployment ID : ' + uniqueID);
     var deleteLogFiles = false;
-
+    var fileExtension : string = isLinux ? '.sh' : '.cmd';
+ 
     if(appOfflineFlag) {
         await appOfflineKuduService(publishingProfile, physicalPath, true);
     }
     try {
-        var mainCmdFilePath = path.join(__dirname, 'postDeploymentScript', 'mainCmdFile.cmd');
-        await uploadFiletoKudu(publishingProfile, physicalPath, 'mainCmdFile_' + uniqueID + '.cmd', mainCmdFilePath);
-        await uploadFiletoKudu(publishingProfile, physicalPath, 'kuduPostDeploymentScript_' + uniqueID + '.cmd', scriptFile.filePath);
+        var mainCmdFilePath = path.join(__dirname, 'postDeploymentScript', 'mainCmdFile' + fileExtension);
+        await uploadFiletoKudu(publishingProfile, physicalPath, 'mainCmdFile_' + uniqueID + fileExtension, mainCmdFilePath);
+        await uploadFiletoKudu(publishingProfile, physicalPath, 'kuduPostDeploymentScript_' + uniqueID + fileExtension, scriptFile.filePath);
         console.log(tl.loc('ExecuteScriptOnKudu', publishingProfile.publishUrl));
-        await runCommandOnKudu(publishingProfile, physicalPath, 'mainCmdFile_' + uniqueID + '.cmd ' + uniqueID, 30, 'script_result_' +  uniqueID + '.txt');
+        if(isLinux){
+            await runCommandOnKudu(publishingProfile, physicalPath, 'sh ' + 'mainCmdFile_' + uniqueID + fileExtension + ' ' + uniqueID, 30, 'script_result_' +  uniqueID + '.txt');
```

Comment:
maybe you can combine the isLinux check at the top and assign file name to variable instead of checking twice ( while deciding file extension and here)

Reply:
In that case I will have to assign 3 filenames ( for mainCmdFile, kuduDeploymentScript and delete_log_file). Also the base name of the files remain same, so I avoided doing that.